### PR TITLE
Fix and reenable some deposit processing tests

### LINKED ISSUE #2375: Reinstate tests in BlockProcessorUtilTest and add test for bad public key
Both of the existing deposit processing tests in _BlockProcessorUtilTest_ for `process_deposits()` are disabled and should be fixed and re-enabled.

In addition, we should add a test that ensures that deposits with a bad public key (specifically not in G1) are cleanly ignored.

### PR DESCRIPTION
**Pull request by benjaminion**
## Description

1. Fix and reenable a couple of deposit processing tests by adding deposit contract Merkle root calculation logic.
2. Add a test to ensure that a deposit with an invalid public key (not in G1) is handled correctly (i.e. silently ignored).

Resolves #2375

## Documentation

- [x ] I thought about documentation and added the `documentation` label to this PR if updates are required.

### CODE REVIEW

File: ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/util/BlockProcessorUtilTest.java [hunk 1]
```diff
@@ -42,18 +46,31 @@
...
+            .updated(
+                state ->
+                    state.setEth1_data(
+                        new Eth1Data(
+                            depositMerkleTree.getRoot(), UnsignedLong.valueOf(1), Bytes32.ZERO)));
+
+    SSZMutableList<DepositWithIndex> deposits =
+        SSZList.createMutable(DepositWithIndex.class, Constants.MAX_DEPOSITS);
+    deposits.add(
+        new DepositWithIndex(depositMerkleTree.getProof(0), depositInput, UnsignedLong.valueOf(0)));
```
Comment (mbaxter) [inline ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/util/BlockProcessorUtilTest.java]: (optional) Might be worth trying to dedupe some of this logic with a helper method that sets up and processes the state & deposits and returns the resulting state.

File: ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/util/BlockProcessorUtilTest.java [hunk 2]
```diff
@@ -127,6 +156,56 @@ void processDepositTopsUpValidatorBalanceWhenPubkeyIsFoundInRegistry()
...
+    // Attempt to process deposit with above data. We expect to fail, but not throw an exception.
+    beaconState =
+        beaconState.updated(state -> BlockProcessorUtil.process_deposits(state, deposits));
+
+    assertTrue(
+        beaconState.getValidators().size() == originalValidatorRegistrySize,
+        "The validator was added to the validator registry.");
+    assertTrue(
+        beaconState.getBalances().size() == originalValidatorBalancesSize,
+        "The balance was added to the validator balances.");
```
Comment (mbaxter) [inline ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/util/BlockProcessorUtilTest.java:140]: (optional) Maybe we should verify that all of the balances are unchanged.

**Review by mbaxter — APPROVED** [review]

File: ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/util/BlockProcessorUtilTest.java (hunk 2 above)
Comment (benjaminion) [reply ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/util/BlockProcessorUtilTest.java:140]: Good idea. Done.

File: ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/util/BlockProcessorUtilTest.java (hunk 1 above)
Comment (benjaminion) [reply ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/util/BlockProcessorUtilTest.java]: Heh, busted. Yes, I was too lazy to do this. Now done.

**Pull request merged**